crates/ty_python_semantic/src/site_packages.rs

```diff
@@ -298,10 +275,23 @@ impl VirtualEnvironment {
         // created the `pyvenv.cfg` file. Lenient parsing is appropriate here:
         // the file isn't really *invalid* if it doesn't have this key,
         // or if the value doesn't parse according to our expectations.
-        let version = version_info_string.and_then(|version_string| {
+        let version = version.and_then(|(version_string, range)| {
             let mut version_info_parts = version_string.split('.');
             let (major, minor) = (version_info_parts.next()?, version_info_parts.next()?);
-            PythonVersion::try_from((major, minor)).ok()
+            let version = PythonVersion::try_from((major, minor)).ok()?;
+
+            let canonical_path = pyvenv_cfg_path
+                .as_std_path()
+                .canonicalize()
+                .ok()
+                .and_then(|path| SystemPathBuf::from_path_buf(path).ok())
+                .map(Arc::new);
```

Comment:
What's the reason that we need to canonicalize here. Could we use `absolute` instead?

Also, `as_std_path` won't work with our memory file system. Use `System::canonicalize` instead

Reply:
Where did you see this. Where does the `cwd` directory come from? We don't canonicalize it in the CLI

Reply:
I think all our other paths must already be UNC paths on Windows (for some reason), so I think a lot of other things would start failing if `cwd` wasn't canonicalized. (But I'm also happy to look again and find a different fix!)

Reply:
We do have to canonicalize in the CLI tests because of macos (thanks for that!) You could try to use `dunce::simplified` to simplify the path or use `System::canonicalize_path` although I'm not sure if the docs are accurate because I don't see the dunce::simplified :D

> Hmm, our docs for System::canonicalize_path() state:

Reply:
Sorry, I'm wrong. we actually do... 

https://github.com/astral-sh/ruff/blob/9ae698fe30cf3526f0e7ae237b800b3ed19a819f/crates/ruff_db/src/system/os.rs#L87-L91

We call simplified. You could try to add the simplified call in the CLI test

Reply:
Okay cool. I'll trust you that this is purely an issue with our test setup and not something that Windows users will actually run into ;)

Reply:
Fixed in https://github.com/astral-sh/ruff/pull/18057/commits/06bd1a58054ac2e53ddf2e566f8015b0f9a72a75 -- ty for pointing me towards the correct fix here!!